src/plugins/dashboard/public/dashboard_container/embeddable/create/search_sessions/start_dashboard_search_session_integration.ts

```diff
@@ -54,38 +53,28 @@ export function startDashboardSearchSessionIntegration(
     ?.pipe(skip(1))
     .subscribe(() => this.forceRefresh());
 
-  // listen to and compare states to determine when to launch a new session.
-  this.getInput$()
-    .pipe(pairwise())
-    .subscribe((states) => {
-      const [previous, current] = states as DashboardContainerInput[];
-      const shouldRefetch = getShouldRefresh.bind(this)(previous, current);
-      if (!shouldRefetch) return;
+  newSession$(this).subscribe(() => {
+    const currentSearchSessionId = this.getState().explicitInput.searchSessionId;
 
-      const currentSearchSessionId = this.getState().explicitInput.searchSessionId;
-
-      const updatedSearchSessionId: string | undefined = (() => {
-        // do not update session id if this is irrelevant state change to prevent excessive searches
-        if (!shouldRefetch) return;
-
-        let searchSessionIdFromURL = getSearchSessionIdFromURL();
-        if (searchSessionIdFromURL) {
-          if (session.isRestore() && session.isCurrentSession(searchSessionIdFromURL)) {
-            // we had previously been in a restored session but have now changed state so remove the session id from the URL.
-            removeSessionIdFromUrl();
-            searchSessionIdFromURL = undefined;
-          } else {
-            session.restore(searchSessionIdFromURL);
-          }
+    const updatedSearchSessionId: string | undefined = (() => {
+      let searchSessionIdFromURL = getSearchSessionIdFromURL();
```

Comment:
Something strange I've noticed here is that we don't ever subscribe to changes in the session ID in the URL. This means that if the session gets removed, then the user hits the back button, we don't actually restore that ID again.

This won't block this PR because this behaviour is present in main also - just something strange to note.